src/libcore/ptr.rs

```diff
@@ -370,6 +680,30 @@ pub unsafe fn read_unaligned<T>(src: *const T) -> T {
 ///     assert_eq!(std::ptr::read(y), 12);
 /// }
 /// ```
+///
+/// Manually implement [`mem::swap`]:
+///
+/// ```
+/// use std::ptr;
+///
+/// fn swap<T>(a: &mut T, b: &mut T) {
+///     unsafe {
+///         let tmp = ptr::read(a);
+///         ptr::copy_nonoverlapping(b, a, 1);
+///         ptr::write(b, tmp);
```

Comment:
This doesn't have the extensive comments that the version on `read` had, and it also has the problem of dropping `tmp`.

Reply:
I'll add some more comments, but again, `write` takes its second parameter by value.